modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -216,7 +219,7 @@ object ConsensusStateUpdater {
                     Applicative[F].whenA(majorityTrigger === TimeTrigger)(consensusStorage.clearTimeTrigger) >>
                       state.facilitators.hash.flatMap { facilitatorsHash =>
                         for {
-                          peerEvents <- consensusStorage.pullEvents(bound)
+                          peerEvents <- consensusStorage.pullEvents(bound) >>= eventCutter.cut
```

Comment:
@kpudlik @TheMMaciek @ryle-goehausen 
We can do this instead,
```
peerEvents <- consensusStorage.pullEvents(bound)
reducedPeerEvents <- eventCutter.cut(peerEvents)
```
then I'll need some guidance on where each of the results should be plugged in.

Reply:
@sdavidp can you move the event cutter to run inside `createProposalArtifact`? 🤔 so basically "within" pull events

Reply:
In general, we're expecting dropped events in other places (particularly regarding rejections and re-submissions elsewhere,) so it's okay to maintain that for now. If it can be easily fixed by moving it into the pull function, then that makes sense to change and implement it here, as there's no reason (yet) to drop them arbitrarily (although there will be in the future due to memory constraints on the pool.)

Reply:
> we're expecting dropped events in other places 

@ryle-goehausen can you elaborate on that part? afaik, we handle all the places programatically to return the discarded events back to the pool (unless they are clearly invalid)

Reply:
Can discuss in meeting, it was referenced in prior ticket for metagraph related rejections I believe? There was a prior workaround done by marcus for this.